server/server.go

```diff
@@ -470,6 +470,13 @@ func (s *Server) processPublish(cl *clients.Client, pk packets.Packet) error {
 		return nil
 	}
 
+	// if an OnProcessMessage hook exists, potentially modify the packet.
+	if s.Events.OnProcessMessage != nil {
+		if pkx, err := s.Events.OnProcessMessage(cl.Info(), events.Packet(pk)); err == nil {
+			pk = packets.Packet(pkx)
```

Comment:
That seems also missing for `OnMessage`

Reply:
Interesting--I missed that. I guess I don't see why the callback needs to return an error if we're not going to display it. If the user wanted to silently not modify the packet, they could just return the original packet.

So, I guess I'm interested in `onError()` calls in both places :)

Reply:
I generally agree with the above. One question I'd pose is whether it makes sense to log any errors which occurred from this event using OnError, or whether we are relying on the embedding service to handle error logging from the function separately. 

I can see a use case where you may wish to drop a packet if it does not satisfy custom logic within `OnProcessMessage`. This could be achieved by returning a new `ErrDropPacket = fmt.Errorf("packet dropped: %w", err)` error value, and then checking for it on line 475:

```go
if s.Events.OnProcessMessage != nil {
  pkx, err := s.Events.OnProcessMessage(cl.Info(), events.Packet(pk));
  if err == nil {
      pk = packets.Packet(pkx)
  } else {
      if err == ErrDropPacket {
        return nil
      }
      // OnError log
  }
}
```
Or something thereabouts. Thoughts?

Reply:
The use of `ErrDropPacket` works for me.